From e8e0fff7b9df8b05a921905ede3a9610b4b62dbb Mon Sep 17 00:00:00 2001
From: Enis Inan <enis.inan@puppet.com>
Date: Tue, 11 Dec 2018 01:08:02 -0800
Subject: [PATCH] [PATCH] Revert "build.info: adapt to the new handling of .rc
 / .def / .map / .opt files"

This reverts commit 7a061312038ce909a214edc4e21c8981c77bc650. We need to
do this when building OpenSSL 1.1.1 for Solaris 10 SPARC as otherwise,
the build will fail with a message indicating that the generated .map
files (e.g. like libcrypto.map) do not have the right format.
---
 build.info | 57 ++-------------------------------------------------------
 1 file changed, 2 insertions(+), 55 deletions(-)

diff --git a/build.info b/build.info
index 3dda4e89bf..21be1071de 100644
--- a/build.info
+++ b/build.info
@@ -9,6 +9,8 @@
      "";
 -}
 LIBS=libcrypto libssl
+ORDINALS[libcrypto]=crypto
+ORDINALS[libssl]=ssl
 INCLUDE[libcrypto]=. include
 INCLUDE[libssl]=. include
 DEPEND[libssl]=libcrypto
@@ -24,61 +26,6 @@
 DEPEND[include/crypto/dso_conf.h]=configdata.pm
 GENERATE[include/crypto/dso_conf.h]=include/crypto/dso_conf.h.in
 
-IF[{- defined $target{shared_defflag} -}]
-  IF[{- $config{target} =~ /^mingw/ -}]
-    GENERATE[libcrypto.def]=util/mkdef.pl crypto 32
-    DEPEND[libcrypto.def]=util/libcrypto.num
-    GENERATE[libssl.def]=util/mkdef.pl ssl 32
-    DEPEND[libssl.def]=util/libssl.num
-
-    SHARED_SOURCE[libcrypto]=libcrypto.def
-    SHARED_SOURCE[libssl]=libssl.def
-  ELSIF[{- $config{target} =~ /^aix/ -}]
-    GENERATE[libcrypto.map]=util/mkdef.pl crypto aix
-    DEPEND[libcrypto.map]=util/libcrypto.num
-    GENERATE[libssl.map]=util/mkdef.pl ssl aix
-    DEPEND[libssl.map]=util/libssl.num
-
-    SHARED_SOURCE[libcrypto]=libcrypto.map
-    SHARED_SOURCE[libssl]=libssl.map
-  ELSE
-    GENERATE[libcrypto.map]=util/mkdef.pl crypto linux
-    DEPEND[libcrypto.map]=util/libcrypto.num
-    GENERATE[libssl.map]=util/mkdef.pl ssl linux
-    DEPEND[libssl.map]=util/libssl.num
-
-    SHARED_SOURCE[libcrypto]=libcrypto.map
-    SHARED_SOURCE[libssl]=libssl.map
-  ENDIF
-ENDIF
-# VMS and VC don't have parametrised .def / .symvec generation, so they get
-# special treatment, since we know they do use these files
-IF[{- $config{target} =~ /^VC-/ -}]
-  GENERATE[libcrypto.def]=util/mkdef.pl crypto 32
-  DEPEND[libcrypto.def]=util/libcrypto.num
-  GENERATE[libssl.def]=util/mkdef.pl ssl 32
-  DEPEND[libssl.def]=util/libssl.num
-
-  SHARED_SOURCE[libcrypto]=libcrypto.def
-  SHARED_SOURCE[libssl]=libssl.def
-ELSIF[{- $config{target} =~ /^vms/ -}]
-  GENERATE[libcrypto.opt]=util/mkdef.pl crypto "VMS"
-  DEPEND[libcrypto.opt]=util/libcrypto.num
-  GENERATE[libssl.opt]=util/mkdef.pl ssl "VMS"
-  DEPEND[libssl.opt]=util/libssl.num
-
-  SHARED_SOURCE[libcrypto]=libcrypto.opt
-  SHARED_SOURCE[libssl]=libssl.opt
-ENDIF
-
-IF[{- $config{target} =~ /^(?:Cygwin|mingw|VC-)/ -}]
-  GENERATE[libcrypto.rc]=util/mkrc.pl libcrypto
-  GENERATE[libssl.rc]=util/mkrc.pl libssl
-
-  SHARED_SOURCE[libcrypto]=libcrypto.rc
-  SHARED_SOURCE[libssl]=libssl.rc
-ENDIF
-
 IF[{- $config{target} =~ /^Cygwin/ -}]
  SHARED_NAME[libcrypto]=cygcrypto-{- $sover_filename -}
  SHARED_NAME[libssl]=cygssl-{- $sover_filename -}
-- 
2.15.1

